saleor/account/notifications.py

```diff
@@ -0,0 +1,121 @@
+
+    user_payload = {
+        "user": get_default_user_payload(user),
+        "recipient_email": user.email,
+        "token": token,
+        "reset_url": reset_url,
+        **get_site_context(),
+    }
+
+    event = (
+        NotifyEventType.ACCOUNT_STAFF_RESET_PASSWORD
+        if staff
+        else NotifyEventType.ACCOUNT_PASSWORD_RESET
+    )
+    manager.notify(event, payload=user_payload)
+
+
+def send_account_confirmation(user, redirect_url, manager):
+    """Trigger sending an account confirmation notification for the given user."""
+    token = default_token_generator.make_token(user)
+    params = urlencode({"email": user.email, "token": token})
+    confirm_url = prepare_url(params, redirect_url)
+    payload = {
+        "user": get_default_user_payload(user),
+        "recipient_email": user.email,
+        "token": token,
+        "confirm_url": confirm_url,
+        **get_site_context(),
+    }
+    manager.notify(NotifyEventType.ACCOUNT_CONFIRMATION, payload=payload)
+
+
+def send_request_user_change_email_notification(
+    redirect_url, user, new_email, token, manager
+):
+    """Trigger sending a notification change email for the given user."""
+    params = urlencode({"token": token})
+    redirect_url = prepare_url(params, redirect_url)
+    payload = {
+        "user": get_default_user_payload(user),
+        "recipient_email": new_email,
+        "old_email": user.email,
+        "new_email": new_email,
+        "token": token,
+        "redirect_url": redirect_url,
+        **get_site_context(),
+    }
+    manager.notify(NotifyEventType.ACCOUNT_CHANGE_EMAIL_REQUEST, payload=payload)
+
+
+def send_user_change_email_notification(recipient_email, user, manager):
+    """Trigger sending a email change notification for the given user."""
+    payload = {
+        "user": get_default_user_payload(user),
+        "recipient_email": recipient_email,
+        **get_site_context(),
+    }
+    manager.notify(NotifyEventType.ACCOUNT_CHANGE_EMAIL_CONFIRM, payload=payload)
+
+
+def send_account_delete_confirmation_notification(redirect_url, user, manager):
+    """Trigger sending a account delete notification for the given user."""
+    token = default_token_generator.make_token(user)
+    params = urlencode({"token": token})
+    delete_url = prepare_url(params, redirect_url)
+    payload = {
+        "user": get_default_user_payload(user),
+        "recipient_email": user.email,
+        "token": token,
+        "delete_url": delete_url,
+        **get_site_context(),
+    }
+    manager.notify(NotifyEventType.ACCOUNT_DELETE, payload=payload)
+
+
+def send_set_password_notification(redirect_url, user, manager, staff=False):
+    """Trigger sending a set password notification for the given customer/staff."""
+    token = default_token_generator.make_token(user)
+    params = urlencode({"email": user.email, "token": token})
+    password_set_url = prepare_url(params, redirect_url)
+    user_payload = {
```

Comment:
🚔  In some functions, this is called `user_payload` and in some other `payload`, while this is more or less the same object, can we unify the naming?